content-security-policy/script-src/script-src-strict_dynamic_discard_whitelist.html

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <title>Whitelists are discarded with `strict-dynamic` in the script-src directive.</title>
+    <script src='/resources/testharness.js' nonce='dummy'></script>
+    <script src='/resources/testharnessreport.js' nonce='dummy'></script>
+    <script src='../support/checkSecurityPolicyViolations.sub.js' nonce='dummy'></script>
+    <!-- CSP served: script-src 'self' 'strict-dynamic' 'nonce-dummy' -->
+</head>
+
+<body>
+    <h1>Whitelists are discarded with `strict-dynamic` in the script-src directive.</h1>
+    <div id='log'></div>
+
+    <script nonce='dummy'>
+        var simpleSourcedScriptRan = false;
+    </script>
+    <script src='simpleSourcedScript.js'></script>
+    <script nonce='dummy'>
+        async_test(function(t) {
+            assert_false(simpleSourcedScriptRan);
+            t.done();
+        }, 'Whitelisted script without a correct nonce is not allowed with `strict-dynamic`.');
+    </script>
+
+    <script nonce='dummy'>
+        async_test(function(t) {
+            setTimeout(t.step_func_done(function() {
+                assert_violations({
+                    'script-src': 1
+                });
+            }), 1000);
```

Comment:
Ditto.

I'd prefer for this to be structured as a single test. That is:

```
<script nonce="dummy">
  async_test(function(t) {
    document.addEventListener("securitypolicyviolation", t.step_func_done(function (e) {
      assert_false(simpleSourcedScriptRan);
      assert_equals(e.violatedDirective, "script-src");
    }));
  });
</script>
```

That captures both the tests you're running in a way that doesn't involve timeouts.